### PR TITLE
Add GUC for foreign key propagation

### PR DESCRIPTION
**Pull request by svenklemm**
Allow controlling FK query behaviour with GUC
enable_foreign_key_propagation.

Disable-check: force-changelog-file

### CODE REVIEW

**Review by akuzm — APPROVED** [review]

**Review by pallavisontakke — COMMENTED** [review]
Do we need an entry in `.unreleased` folder for this change?

**Comment by svenklemm** [comment]
> Do we need an entry in `.unreleased` folder for this change?

I dont think we do since this is not a change in behaviour that needs documentation

File: src/guc.c [hunk 1]
```diff
@@ -65,6 +65,7 @@ bool ts_guc_enable_constraint_exclusion = true;
 bool ts_guc_enable_qual_propagation = true;
 bool ts_guc_enable_cagg_reorder_groupby = true;
 bool ts_guc_enable_now_constify = true;
+bool ts_guc_enable_foreign_key_propagation = true;
```
Comment (erimatnor) [inline src/guc.c:68]: I thought the decision was to not have a GUC and that the default would be to not propagate foreign keys. Or, is this something different?

Comment (svenklemm) [reply src/guc.c:68]: the guc is for support/debugging for foreign key propagation in general, it will not alter tiered storage behaviour

**Review by fabriziomello — APPROVED** [review]

**Pull request merged**